### PR TITLE
feat: 🎸 [JIRA: HCPSDKFIORIUIKIT-2348] shadow effect api

### PR DESCRIPTION
**Pull request by xiaoyu0722**
New API `func shadow(_ style: FioriShadowStyle) -> some View`

### CODE REVIEW

**Comment by xiaoyu0722** [comment]
Any suggestions for shadow api are welcome.

File: Sources/FioriSwiftUICore/Utils/ShadowEffect.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    /// - Returns: A new view with the specified shadow style.
+    func shadow(_ style: FioriShadowStyle) -> some View {
+        self.modifier(FioriShadowModifier(style: style))
+    }
+}
+
+struct FioriShadowModifier: ViewModifier {
+    let style: FioriShadowStyle
+    
+    let color1 = Color(white: 0, opacity: 0.13)
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/Utils/ShadowEffect.swift:35]: Does the spec say the color used here should be "white" ?  I think opacity on top of a "white" color may not create any visual effect.   Also, we need to consider dark mode. So base color should be a dynamic color , I assume. You can follow up with Designer on this.

**Review by dyongxu — APPROVED** [review]

**Review by dyongxu — CHANGES_REQUESTED** [review]
Shadow effect is almost invisible.

File: Sources/FioriSwiftUICore/Utils/ShadowEffect.swift (hunk 1 above)
Comment (xiaoyu0722) [reply Sources/FioriSwiftUICore/Utils/ShadowEffect.swift:35]: Yes, all shadows are black(0 white) with opacity and same values for light and dark mode.
Screenshot 2024-09-06 at 07 54 05

**Review by billzhou0223 — APPROVED** [review]

File: Sources/FioriSwiftUICore/Utils/ShadowEffect.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    case medium
+    /// Heavy shadow. Most used in toast message or quick sort.
+    case heavy
+    /// Max shadow. Most used in popovers.
+    case max
+    
+    /// Compact shadow. Most used in switch or slider.
+    case compact
+    /// Broad shadow. Most used in segmented control.
+    case broad
```
Comment (JuliaSchnSAP) [inline Sources/FioriSwiftUICore/Utils/ShadowEffect.swift:20]: I suggest making the following changes to the case names to match the design guidelines more closely:
- `minimal`...`max` -> `level0`...`level4`
- `compact` -> `smallElement`
- `broad` -> `mediumElement`

File: Sources/FioriSwiftUICore/Utils/ShadowEffect.swift [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                .shadow(color: self.color2, radius: 2, y: 1)
+        case .medium:
+            content
+                .shadow(color: self.color1, radius: 1)
+                .shadow(color: self.color2, radius: 4, y: 2)
+        case .heavy:
+            content
+                .shadow(color: self.color1, radius: 2)
+                .shadow(color: self.color2, radius: 16, y: 8)
+                .shadow(color: self.color2, radius: 32, y: 16)
```
Comment (JuliaSchnSAP) [inline Sources/FioriSwiftUICore/Utils/ShadowEffect.swift:58]: The radius values for this shadow type should be 1, 8 and 16 according to the design spec

File: Sources/FioriSwiftUICore/Utils/ShadowEffect.swift [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                .shadow(color: self.color2, radius: 2, y: 1)
+        case .level2:
+            content
+                .shadow(color: self.color1, radius: 1)
+                .shadow(color: self.color2, radius: 4, y: 2)
+        case .level3:
+            content
+                .shadow(color: self.color1, radius: 1)
+                .shadow(color: self.color2, radius: 16, y: 8)
+                .shadow(color: self.color2, radius: 32, y: 16)
```
Comment (JuliaSchnSAP) [inline Sources/FioriSwiftUICore/Utils/ShadowEffect.swift:58]: To clarify my last comment I meant to change these two lines to the following code:
```
.shadow(color: self.color2, radius: 8, y: 8)
.shadow(color: self.color2, radius: 16, y: 16)
```

Comment (xiaoyu0722) [reply Sources/FioriSwiftUICore/Utils/ShadowEffect.swift]: Sorry my mistake, all fixed.

**Comment by xiaoyu0722** [comment]
@dyongxu Need a review again.

**Review by dyongxu — APPROVED** [review]

**Pull request merged**